source/codegen/service_helpers.py

```diff
@@ -246,7 +246,9 @@ def _create_param(parameter, expand_varargs=True, repeated_parameters=None):
             return "..."
     elif common_helpers.is_array(type):
         array_size = _get_array_param_size(parameter)
-        return f"{type[:-2]} {name}[{array_size}]"
+        if type[:-2] == "void":
```

Comment:
This is a pretty confusing change, I at least understand what it's doing. Is there a reason that `void[]` doesn't work? I know that's unconventional, but it does match how grpc-device usually deals with buffer types.

At least put a comment here on what it's doing and why.